docs/GettingStartedDocs/Contributors/SignUsingEngines.md

```diff
@@ -0,0 +1,114 @@
+  known vulnerabilities or sub-optimal performances with more appropriate alternative implementations while maintaining compatibility
+  with existing applications.
+
+- They provide an alternative in case of issues with the OpenSSL core development team or distro vendor decision-making process, decoupling
+  the decision to adopt the OpenSSL library from the choice of individual cryptosystem implementations, without incurring the 
+  costs of maintaining a fork of the OpenSSL project or patchsets, while also providing transparent binary compatibility with existing applications; 
+
+- They allow hardware vendors to release self-contained plugins to leverage advanced hardware concepts, such as GPU compute, FPLAs, or
+  quantum based encrytion which can work then with existing applications based on OpenSSL, keeping their software outside
+ of the main OpenSSL codebase.
+
+- They allow backporting newer crypto systems in previous versions of the OpenSSL library and existing applications based on it.
+
+- They allow a straightforward method of protyping and deploying new crypto systems or new implementations for already compiled-in 
+  cryptosystems to the OpenSSL library, providing convenient way to test and benchmark new software implementations in a real-world context.
+
+- They offer a greater degree of freedom from the OpenSSL project toolchain, allowing developers to use different programming
+  languages and build systems, potentially lowering the development and maintenance costs for developing plugin alternative
+  implementations or new functionality.
+
+- They offer flexibility to solve licensing issues. 
+
+  Currently the OpenSSL project is released under a "dual license" scheme, under the OpenSSL License,
+  (a derivative of the Apache License 1.0) and the SSLeay License (similar to a 4-clause BSD License). OpenSSL is in the process of 
+  transitioning to the Apache License 2.0. Contributors are thus forced to release their work under these licenses, which may conflict
+  with other applicable licenses especially when reusing code from projects released under a proprietary license or an incompatible copyleft license. 
+
+  Being objects dynamically loaded at runtime, engines can benefit from usually more flexible licensing requirements, providing bridge 
+  towards software released under different licenses.
+
+## Engine internal operation
+
+The cryptographic functionality that can be provided by an engine implementation includes the following abstractions:
+(1) RSA_METHOD, DSA_METHOD, DH_METHOD, EC_METHOD; providing alternative RSA/DSA/etc. implementations; 
+(2) RAND_METHOD: providing alternative (pseudo-)random number generation implementations;
+(3) EVP_CIPHER: providing alternative (symmetric) cipher algorithms;
+(4) EVP_MD: providing alternative message digest algorithms;
+(5) EVP_PKEY: providing alternative public key algorithms.
+
+At the highest level of abstraction, a dynamic engine can be split into two functional blocks.
```

Comment:
Should "dynamic engine" be "dynamic plugin"?

Reply:
dynamic engine is the correct term and what is used in openssl documentation. Plugin is not a synonym but an adjective.